corehq/apps/cloudcare/static/cloudcare/js/form_entry/form_ui.js

```diff
@@ -313,6 +328,13 @@ hqDefine("cloudcare/js/form_entry/form_ui", function () {
         }
 
         for (let child of json.children) {
+            if (isNPerRowRepeatElement(child)) {
```

Comment:
Could all of this be incorporated as part of the `if (child.type === constants.GROUP_TYPE) ` condition?

Reply:
Do you mean to say put this condition within https://github.com/dimagi/commcare-hq/pull/33979/files/cdbbdeba26cd4e11b9b47677863cf2e6515ac94d#diff-fa78ad0c884f5acb33ba17fb2142ba438de13b9bb043c65b1687111cfca48190R345 ?

Reply:
yes, I think it'll have to be changed a bit but it should be able to be combined

Reply:
Can you specify in little more detail what is intended to be done here? Shouldn't adding style `n-per-row` be done to the repeat group's `Group` before reaching the condition `if (child.type === constants.GROUP_TYPE)` ?

Reply:
you're adding `n-per-row` to the children of the element defined as `n-per-row-repeat` right? I think when we get to `if (child.type === constants.GROUP_TYPE)`, we can also check if it also contains the `n-per-row-repeat` appearance attribute. If so, then we populate its children with the respective `n-per-row`

Reply:
https://github.com/dimagi/commcare-hq/pull/33979/commits/ba593a503571ab1a54b2f59c30982a4d56127423

Reply:
ya that's what i was imaging - does that work with testing?

Reply:
Yes, it is working.